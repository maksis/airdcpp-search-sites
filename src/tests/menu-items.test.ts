import {
	FilelistItemGetter,
	HubMessageHighlightItemGetter,
	PrivateChatMessageHighlightItemGetter,
	QueueBundleItemGetter,
	SearchItemGetter,
} from 'src/search-items';

import { getUrls, itemUrlParser } from './helpers';

describe('Search items', () => {
	test('get bundle items', async () => {
		const urls = await getUrls(
			QueueBundleItemGetter,
      itemUrlParser([1, 2], ''),
		);
		expect(urls).toMatchSnapshot();
	});

	test('get search result items', async () => {
		const urls = await getUrls(
			SearchItemGetter,
			itemUrlParser(['mock-id1', 'mock-id2'], 1)
		);
		expect(urls).toMatchSnapshot();
	});

	test('get filelist items', async () => {
		const urls = await getUrls(
			FilelistItemGetter,
			itemUrlParser([1, 2], 'mock-filelist-id')
		);
		expect(urls).toMatchSnapshot();
	});

	test('hub message highlights', async () => {
		const urls = await getUrls(
			HubMessageHighlightItemGetter,
			itemUrlParser([1, 2], 4)
		);
		expect(urls).toMatchSnapshot();
	});

	test('private chat message highlights', async () => {
		const urls = await getUrls(
			PrivateChatMessageHighlightItemGetter,
			itemUrlParser([1, 2], 'mock-private-chat-id')
		);
		expect(urls).toMatchSnapshot();
	});
});
