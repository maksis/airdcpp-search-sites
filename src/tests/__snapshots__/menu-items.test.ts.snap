// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search items get bundle items 1`] = `
Array [
  Array [
    "http://mock-url1/Mock.Bundle.Directory-TEST",
    "http://mock-url1/Mock.Bundle.Directory-TEST",
  ],
  Array [
    "http://mock-url2/mock%20bundle%20directory",
    "http://mock-url2/mock%20bundle%20directory",
  ],
]
`;

exports[`Search items get filelist items 1`] = `
Array [
  Array [
    "http://mock-url1/Mock.Filelist.Directory-TEST",
    "http://mock-url1/Mock.Filelist.Directory-TEST",
  ],
  Array [
    "http://mock-url2/mock%20filelist%20directory",
    "http://mock-url2/mock%20filelist%20directory",
  ],
]
`;

exports[`Search items get search result items 1`] = `
Array [
  Array [
    "http://mock-url1/Mock.Result.Directory-TEST",
    "http://mock-url1/Mock.Result.Directory-TEST",
  ],
  Array [
    "http://mock-url2/mock%20result%20directory",
    "http://mock-url2/mock%20result%20directory",
  ],
]
`;

exports[`Search items hub message highlights 1`] = `
Array [
  Array [
    "http://mock-url1/mock%2Bmagnet%2Bfile.log",
    "http://mock-url1/mock%2Bmagnet%2Bfile.log",
  ],
  Array [
    "http://mock-url2/mock%2Bmagnet%2Bfile%20log",
    "http://mock-url2/mock%2Bmagnet%2Bfile%20log",
  ],
]
`;

exports[`Search items private chat message highlights 1`] = `
Array [
  Array [
    "http://mock-url1/Mock.Highlight.Text-TEST",
    "http://mock-url1/Mock.Highlight.Text-TEST",
  ],
  Array [
    "http://mock-url2/mock%20highlight%20text",
    "http://mock-url2/mock%20highlight%20text",
  ],
]
`;
